### PR TITLE
Guus/update egui

### PR DESCRIPTION
**Pull request by guusw**
- Update egui to 0.27
- Fix selectable behavior and integrate drag-and-drop
- Add UI.Window Min/Max/Fixed size parameters and made them support Var/Float

### CODE REVIEW

File: shards/tests/hello.shs [hunk 1]
```diff
@@ -1,215 +1,9 @@
...
-["7" "8" "9"] = zs3
-["10" "20" "30" "40"] = zs4
-Zip([zs1 zs2 zs3 zs4 [99 98]]) | Assert.Is([
-  ["1" "4" "7" "10" 99]
-  ["2" "5" "8" "20" 98]]) | Log
-
-true = is-true | IsTrue | Assert.Is(true) | Log
-false = is-false | IsFalse | Assert.Is(true) | Log
\ No newline at end of file
+{
```
Comment (guusw) [inline shards/tests/hello.shs]: Ooops

File: docs/samples/shards/UI/AutoGrid/1.shs [hunk 2]
```diff
@@ -7,7 +7,7 @@ GFX.MainWindow(
     UI(
       UI.Window(
         Title: "Greetings"
-        Width: 200 ; Makes window resizable but with a minimum width of 200, allowing resizing window to manipulate number of rows in autogrid
+        MinSize: @f2(200.0 -1.0) ; Makes window resizable but with a minimum width of 200, allowing resizing window to manipulate number of rows in autogrid
```
Comment (sinkingsugar) [inline docs/samples/shards/UI/AutoGrid/1.shs]: what's -1?

File: shards/egui/src/containers/selectable.rs [hunk 3]
```diff
@@ -172,70 +184,128 @@ impl Shard for Selectable {
...
+    // ui.interact(rect, id, sense)
+
+    let drag_data = self.drag_data.get();
+    let can_drag = !drag_data.is_none();
+
+    let mut err: Option<&str> = None;
+    ui.push_id(id, |ui| -> Result<(), &str> {
+      let mut body = |ui: &mut egui::Ui| {
+        // Draw frame and contents
+        with_possible_panic(|| {
```
Comment (sinkingsugar) [inline shards/egui/src/containers/selectable.rs:196]: I guess we try remove those after merge?

File: shards/egui/src/containers/window.rs [hunk 4]
```diff
@@ -52,14 +47,24 @@ struct WindowShard {
   pub position: ParamVar,
   #[shard_param("Anchor", "Corner or center of the screen.", ANCHOR_TYPES)]
   pub anchor: ParamVar,
-  #[shard_param("Width", "The width of the rendered window.", INT_OR_NONE_TYPES_SLICE)]
```
Comment (sinkingsugar) [inline shards/egui/src/containers/window.rs:55]: hmm I think a fixed width should always be present if wanted no?

File: shards/tests/hello.shs (hunk 1 above)
Comment (sinkingsugar) [reply shards/tests/hello.shs]: fix

**Review by sinkingsugar — CHANGES_REQUESTED** [review]

File: docs/samples/shards/UI/AutoGrid/1.shs (hunk 2 above)
Comment (guusw) [reply docs/samples/shards/UI/AutoGrid/1.shs]: No constraint

File: shards/egui/src/containers/selectable.rs (hunk 3 above)
Comment (guusw) [reply shards/egui/src/containers/selectable.rs:196]: I didn't fix the stack thing yet

**Comment by guusw** [comment]
@ShahFragcolor Make sure to update window Width/Height to MinWidth/MinHeight in your scripts, and make them float

**Pull request merged**